sdk/eventhub/event-hubs/src/eventDataBatch.ts

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { EventData, toAmqpMessage } from "./eventData";
+import { ConnectionContext } from "./connectionContext";
+import { AmqpMessage } from "@azure/core-amqp";
+import { messageProperties, message } from "rhea-promise";
+
+/**
+ * A helper class for creating a batch of events; taking into account the max size limit, so that the `EventDataBatch`
+ * can be passed to the Send method of the `Sender` to send the `EventData` objects as a batch messages.
+ *
+ * Use the `tryAdd` function on the EventDataBatch to add events in a batch.
+ * @class
+ */
+export class EventDataBatch {
+  /**
+   * @property Describes the amqp connection context for the Client.
+   */
+  private _context: ConnectionContext;
+  /**
+   * @property Array of event data objects.
+   */
+  events: EventData[] = [];
```

Comment:
For extra safety, make this a public getter backed by a private `_events` field. That way it's clear this a readonly property and someone doesn't try to just set the `events` field to an array of events.